sdk/iot/modelsrepository/src/modelsRepositoryClient.ts

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) Microsoft.
+// Licensed under the MIT license.
+
+import { 
+  GetModelsOptions, 
+  IoTModelsRepositoryServiceClient, 
+  ModelsRepositoryClientOptions,
+  dependencyResolutionType,
+  HttpFetcher,
+  FilesystemFetcher,
+  PseudoParser,
+  DtmiResolver,
+  logger,
+  DTDL,
+} from "./internal";
+import * as cnst from './constants';
+import { createClientPipeline, InternalClientPipelineOptions } from "@azure/core-client";
+import { URL } from "url";
+import * as path from "path";
+import { Fetcher } from "./fetcherAbstract";
+import { RestError } from "@azure/core-rest-pipeline";
+
+function isLocalPath(p: string): boolean {
+  const myRegex = RegExp(/^(?:[a-zA-Z]\:|\\\\[\w\.]+\\[\w.$]+)\\(?:[\w]+\\)*\w([\w.])+$/g);
```

Comment:
In windows, I think network drives can start with the prefix `//` (at least, that's what I see when using explorer.exe to view my wslv2 filesystem).

Maybe it would be simpler to use the `path` module's `isAbsolute` method since the user needs to give you an absolute path anyway. Note that this is only available in node, so you'd want to move this function to a separate file.

Reply:
let's sync via teams chat about what you mean by this. Unfortunately my understanding of shims is at a limit here.